Automattic/wp-calypso
# Framework: Get rid of impure-lodash

#### Changes proposed in this Pull Request

We've been relying on re-exporting several impure functions of Lodash to allow for easier mocking in tests, but since that's no longer necessary, this PR is removing those. Furthermore, we use the opportunity to remove some `uniqueId` Lodash usage in favor of `uuid.v4()` which we already use in many places. As a next step, we can remove the rest of the `uniqueId` instances from Lodash completely, but that's work for another PR (done in #52673).

If you want to find out why we're moving away from Lodash, see https://github.com/Automattic/wp-calypso/pull/50368 and p4TIVU-9Bf-p2.

#### Testing instructions

* Verify all tests pass.
* Smoke test media, including uploading of a new image and verify everything still works well.
